### PR TITLE
[MI-1817] add chat visitor changed event

### PR DESCRIPTION
**Pull request by joseconsador**
/cc @zendesk/mintegrations @eddiesholl 

### Description

Add support for chat - now the app actually works in chat.
Added visitor.email.changed event to the fetchCustomer events.
Refactored some async zafClient calls as action creators.

### Note
Some strings are **not** i18n, this is due to the fact that their keys have just been added and will not exist on the compiled JS until after a string sweep.

### References
* JIRA: https://zendesk.atlassian.net/browse/MI-1817

### Risks
* [Medium] May have some unhandled errors which will not display on screen

### CODE REVIEW

File: src/Actions/CustomerActions.js [hunk 1]
```diff
@@ -1,10 +1,40 @@
...
+    const zafClient = getState().app.zafClient;
+    const location = getState().app.context.location;
+
+    let emailKey = 'ticket.requester.email';
+
+    if (location === 'chat_sidebar') {
+      emailKey = 'visitor.email';
+    } else if (location === 'user_sidebar') {
+      emailKey = 'user.email';
+    } else if (location === 'new_ticket_sidebar') {
```
Comment (samgavinio) [inline src/Actions/CustomerActions.js]: Are there other `location` values that we may get here? Would an `else` statement suffice in that case?

File: tests/Actions/AppActions.test.js [hunk 2]
```diff
@@ -52,13 +62,66 @@ describe('setConfig', () => {
...
+    stubbedGet = sinon.stub(zafClient, 'get');
+  });
+
+  afterEach(() => {
+    stubbedGet.restore();
+  });
+
+  it('should dispatch an action to set the url with the correct subdomain', async () => {
+    const subdomain = faker.random.word();
+    const data = { context: { product: 'support' } };
```
Comment (samgavinio) [inline tests/Actions/AppActions.test.js]: Do we have a test for when the product is `chat`?

File: tests/Reducers/AppReducer.test.js [hunk 3]
```diff
@@ -62,6 +79,7 @@ describe('app reducer', () => {
     Reducer(app).expect(action).toReturnState(expectedState);
   });
 
+
```
Comment (samgavinio) [inline tests/Reducers/AppReducer.test.js]: Extra space here, :trollface:

File: src/Actions/CustomerActions.js (hunk 1 above)
Comment (joseconsador) [reply src/Actions/CustomerActions.js]: These are all the locations that are specified on our manifest (ticket_sidebar included, which maps to the emailKey ticket.requester.email - the default value of `emailKey`). If the app is somehow ran on a different location, an error from the app framework would be displayed.

**Comment by joseconsador** [comment]
@samgavinio 💇‍♂️

File: src/Actions/AppActions.js [hunk 4]
```diff
@@ -27,21 +30,49 @@ export const setConfig = data => ({
...
+    return dispatch({
+      type: SET_TRANSLATIONS,
+      locale,
+      translations,
+    });
+  }
+
+  return zafClient.get('currentUser.locale').then(
+      localeData => loadTranslationsForLocale(localeData['currentUser.locale']),
+      () => loadTranslationsForLocale('en-US'), // Load the default locale, locale is not supported for chat apps
```
Comment (kcasas) [inline src/Actions/AppActions.js]: use `DEFAULT_LOCALE` from `../translations`

**Pull request merged**